controller/controller_server/exception_handler.py

```diff
@@ -0,0 +1,43 @@
+from decorator import decorator
+
+import grpc
+
+import controller.array_action.errors as controller_errors
+from controller.common.csi_logger import get_stdout_logger
+from controller.controller_server.errors import ObjectIdError
+from controller.controller_server.errors import ValidationException
+from controller.csi_general import csi_pb2
+
+logger = get_stdout_logger()
+
+exceptions_to_status_codes_map = {
+    ValidationException: grpc.StatusCode.INVALID_ARGUMENT,
+    ObjectIdError: grpc.StatusCode.NOT_FOUND,
+    controller_errors.IllegalObjectName: grpc.StatusCode.INVALID_ARGUMENT,
+    controller_errors.SpaceEfficiencyNotSupported: grpc.StatusCode.INVALID_ARGUMENT,
+    controller_errors.PoolDoesNotExist: grpc.StatusCode.INVALID_ARGUMENT,
+    controller_errors.PoolDoesNotMatchCapabilities: grpc.StatusCode.INVALID_ARGUMENT,
+    controller_errors.PoolParameterIsMissing: grpc.StatusCode.INVALID_ARGUMENT,
+    controller_errors.ExpectedSnapshotButFoundVolumeError: grpc.StatusCode.INVALID_ARGUMENT,
+    controller_errors.ObjectNotFoundError: grpc.StatusCode.NOT_FOUND,
+    controller_errors.PermissionDeniedError: grpc.StatusCode.PERMISSION_DENIED,
+    controller_errors.VolumeAlreadyExists: grpc.StatusCode.ALREADY_EXISTS,
+    controller_errors.NotEnoughSpaceInPool: grpc.StatusCode.RESOURCE_EXHAUSTED
+}
+
+
+response_types = {
+    "CreateVolume": csi_pb2.CreateVolumeResponse
+}
```

Comment:
I just thought of a way to avoid this dict.
```
def handle_common_exceptions(response_type):
    @decorator
    def common_exceptions_handler(...):
        ...
        except ...:
            ...
            return response_type()
    return common_exceptions_handler
```

```
@handle_common_exceptions(csi_pb2.CreateVolumeResponse)
def CreateVolume(self, request, context):
    ....
```
what do you think?

Reply:
I like it.
I changed the decorator name for clarity purposes.

Reply:
I'm not sure `decorated_handle_common_exceptions` is clearer than `common_exceptions_handler` (or even correct)

Reply:
It's recommended (by naming convention) to name functions like verbs. i.e.:
do_something
start_program
load_configuration

It's recommended (by naming convention) to name classes like nouns. i.e.:
RequestHandler
ConfigurationLoader
NetworkManager

`common_exceptions_handler` is a noun. Thus, not recommended to be a function name.
Why `decorated_handle_common_exceptions` is incorrect?

Reply:
1. I don't see any function name in your examples that starts with an adjective
2. IMO it's more "decorating" than it is "decorated", but the outer function is also "decorating", so I don't see the value of having this term in the name

then how about:
```
def handle_common_exceptions_with(response_type):
    @decorator
    def handle_common_exceptions(...):
        ...
    return handle_common_exceptions
```

Reply:
1. Usually it starts with a verb, not an adjective


`handle_common_exceptions_with` with what?

@ArbelNathan , we have a decorator inside a function and we're struggling with naming the function and the decorated function. Do you have any suggestions?

Reply:
with "the following" response type (not the most common style, but I think it's very readable, especially from the calling side)

Reply:
Done